src/Umbraco.Web/PropertyEditors/ValueConverters/ContentPickerPropertyConverter.cs

```diff
@@ -104,33 +104,26 @@ public override object ConvertDataToSource(PublishedPropertyType propertyType, o
         public override object ConvertSourceToObject(PublishedPropertyType propertyType, object source, bool preview)
         {
             if (source == null)
-            {
                 return null;
-            }
 
-            if (UmbracoContext.Current == null) return source;
+            var umbracoContext = UmbracoContext.Current;
+            if (umbracoContext == null)
+                return source;
+
+            // Do not convert source for excluded properties
+            if (propertyType.PropertyTypeAlias == null || PropertiesToExclude.InvariantContains(propertyType.PropertyTypeAlias))
+                return source;
 
-            if ((propertyType.PropertyTypeAlias != null && PropertiesToExclude.Contains(propertyType.PropertyTypeAlias.ToLower(CultureInfo.InvariantCulture))) == false)
+            if (source is int sourceInt)
             {
-                IPublishedContent content;
-                if (source is int)
-                {
-                    var sourceInt = (int)source;
-                    content = UmbracoContext.Current.ContentCache.GetById(sourceInt);
-                    if(content != null)
-                        return content;
-                }
-                else
-                {
-                    var sourceUdi = source as Udi;
-                    if (sourceUdi == null) return null;
-                    var umbHelper = new UmbracoHelper(UmbracoContext.Current);
-                    content = umbHelper.TypedContent(sourceUdi);
-                    if (content != null)
-                        return content;
-                }
+                return umbracoContext.ContentCache.GetById(sourceInt);
             }
-            return source;
+            else if (source is Udi sourceUdi)
+            {
+                return umbracoContext.ContentCache.GetById(sourceUdi);
+            }
+
+            return null;
```

Comment:
As `ConvertDataToSource` makes sure the source value can only be of type `int` or `Udi`, this makes sure it doesn't return that source value for unpublished/deleted content!